internal/config/system.go

```diff
@@ -56,9 +58,19 @@ func (s *System) DefaultAndValidate() error {
 		s.ModelAutoscaling.TimeWindow.Duration = 10 * time.Minute
 	}
 
+	if s.ModelRollouts.PodReadinessWaitPeriod.Duration == 0 {
+		s.ModelRollouts.PodReadinessWaitPeriod.Duration = 10 * time.Minute
+	}
+
 	return validator.New(validator.WithRequiredStructEnabled()).Struct(s)
 }
 
+type ModelRollouts struct {
+	// PodReadinessWaitPeriod is the time to wait for a Pod to become ready before
+	// progressing to rolling out the next Pod.
+	PodReadinessWaitPeriod Duration `json:"podReadinessWaitPeriod" validate:"required"`
```

Comment:
looks like it's required? Should we set a default in values.yaml?

Reply:
The validation happens after the defaulting... Its like that for a few other fields too. Its kinda confusing - should probably remove.

Reply:
Outdated